### PR TITLE
Implement error returns for listing wrappers

### PR DESCRIPTION
**Pull request by adreed-msft**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] Apache v2 license headers are included in each file.
 
[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/master/CHANGELOG.md

### CODE REVIEW

File: sdk/storage/azblob/zc_list_blobs_wrapper.go [hunk 1]
```diff
@@ -5,25 +5,26 @@ import (
...
+	errChan chan error
 	ctx context.Context
 
 	// Set to 0 for no time-out
 	timeout time.Duration
-	timer *time.Timer
+	timer   *time.Timer
 }
 
 func (p listBlobsFlatSegmentAutoPager) Go() {
```
Comment (zezha-msft) [inline sdk/storage/azblob/zc_list_blobs_wrapper.go:30]: Any test for this?

Comment (adreed-msft) [reply sdk/storage/azblob/zc_list_blobs_wrapper.go:30]: Not yet. Lemme implement one.

**Comment by mohsha-msft** [comment]
Hey @adreed-msft!

Consider merging [this](https://github.com/Azure/azure-sdk-for-go/pull/14174/files) to resolve the conflicts.

File: sdk/storage/azblob/zc_list_blobs_wrapper.go [hunk 2]
```diff
@@ -68,19 +84,32 @@ func (p listBlobsHierarchySegmentAutoPager) Go() {
 			for _, v := range *resp.EnumerationResults.Segment.BlobItems {
```
Comment (mohsha-msft) [inline sdk/storage/azblob/zc_list_blobs_wrapper.go:84]: Hey @adreed-msft ,

The response of pacer we've written in the Listing API(Hierarchical) is a channel of `BlobItemInternal` struct. 

Actual response looks like this. 
```
type ListBlobsHierarchySegmentResponse struct {
   ContainerName   *string                   `xml:"ContainerName,attr"`
   Delimiter       *string                   `xml:"Delimiter"`
   Marker          *string                   `xml:"Marker"`
   MaxResults      *int32                    `xml:"MaxResults"`
   NextMarker      *string                   `xml:"NextMarker"`
   Prefix          *string                   `xml:"Prefix"`
   Segment         *BlobHierarchyListSegment `xml:"Blobs"`
   ServiceEndpoint *string                   `xml:"ServiceEndpoint,attr"`
}
```
```
type BlobHierarchyListSegment struct {
   BlobItems    *[]BlobItemInternal `xml:"Blob"`
   BlobPrefixes *[]BlobPrefix       `xml:"BlobPrefix"`
}
```
I see that we're only iterating over BlobItems so we don't have other attributes in the response. This is problematic as we're returning only a fragment of actual response. Many of our use cases involve other attributes as well.

**Comment by zezha-msft** [comment]
Included in https://github.com/Azure/azure-sdk-for-go/pull/14209.

**Pull request closed without merging**